torchtitan/tools/profiling.py

```diff
@@ -57,10 +57,16 @@ def trace_handler(prof):
         assert (
             wait >= 0
         ), "profile_freq must be greater than or equal to warmup + active"
+        if torch.cuda.is_available():
+            gpu_device_profiled = torch.profiler.ProfilerActivity.CUDA
+        elif torch.xpu.is_available():
+            gpu_device_profiled = torch.profiler.ProfilerActivity.XPU
+        else:
+            raise RuntimeError("Profiling supported only for CUDA and XPU devices")
```

Comment:
If someone is running torchtitan with just CPU wouldn't this throw an error?

Reply:
Yes, my (maybe wrong) motivation was that GPU availability is kind of essential and the CPU-only scenario has not been tested but there is absolutely no reason to restrict it here. Will fix it.